firebase/src/core/data/gateways/filesystem-gateway.ts

```diff
@@ -0,0 +1,38 @@
+import * as fs from "fs";
+import FilesystemError from "@faults/errors/filesystem-error";
+
+/** Exposes common filesystem operations. */
+export class FileSystemGateway {
+  readonly #root = process.cwd();
```

Comment:
This is problematic, since `process.cwd` is relative to the path that the process originates. If this gateways is invoked from `scripts/`, it will have a different path than if invoked through `functions/` as an example. See my general comment below